File: final_dataset_A/generated_code/claude/python/Create_forum_post_023_Secure.py
import html
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def create_forum_post(title, content):
    """
    Let a user create a new forum post (title and content) for a public
    community platform, validating the fields before storing them.
    """
    if not title or not content:
        return "Failure: title and content are required"

    safe_title = html.escape(title.strip()[:200])
    safe_content = html.escape(content.strip()[:5000])

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "INSERT INTO forum_posts (title, content) VALUES (?, ?)",
            (safe_title, safe_content),
        )
        conn.commit()
        return "Post created successfully"
    except Exception:
        return "Failure: unable to create the post at this time"
    finally:
        conn.close()

File: final_dataset_A/generated_code/claude/java/Create_forum_post_023_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class Create_forum_post_023_Secure {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String createForumPost(String title, String content) {
        if (title == null || title.isBlank() || content == null || content.isBlank()) {
            return "Failure: title and content are required";
        }

        String safeTitle = escapeHtml(truncate(title.trim(), 200));
        String safeContent = escapeHtml(truncate(content.trim(), 5000));

        try (Connection conn = getConnection();
             PreparedStatement ps = conn.prepareStatement(
                     "INSERT INTO forum_posts (title, content) VALUES (?, ?)")) {
            ps.setString(1, safeTitle);
            ps.setString(2, safeContent);
            ps.executeUpdate();
            return "Post created successfully";
        } catch (SQLException e) {
            return "Failure: unable to create the post at this time";
        }
    }

    private static String truncate(String value, int maxLength) {
        return value.length() > maxLength ? value.substring(0, maxLength) : value;
    }

    private static String escapeHtml(String value) {
        return value.replace("&", "&amp;")
                .replace("<", "&lt;")
                .replace(">", "&gt;")
                .replace("\"", "&quot;")
                .replace("'", "&#x27;");
    }
}
